qmf/src/libraries/qmfclient/qmailmessage.cpp

```diff
@@ -4169,7 +4179,11 @@ QByteArray QMailMessagePartContainerPrivate::boundary() const
 
 void QMailMessagePartContainerPrivate::setBoundary(const QByteArray& text)
 {
-    _boundary = text;
+    if (text == QByteArray("NIL")) {
```

Comment:
I can move the code here to avoid that concern, in this case NIL wont be a valid boundary: https://github.com/nemomobile-packages/messagingframework/blob/master/qmf/src/plugins/messageservices/imap/imapstructure.cpp#L437